go-controller/pkg/kubevirt/dhcp_test.go

```diff
@@ -0,0 +1,127 @@
+package kubevirt
+
+import (
+	"net"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+)
+
+var _ = Describe("Kubevirt", func() {
+	type dhcpTest struct {
+		cidrs               []string
+		hostname            string
+		dns                 *corev1.Service
+		hasError            bool
+		expectedIPv4Options *nbdb.DHCPOptions
+		expectedIPv6Options *nbdb.DHCPOptions
+	}
+	var (
+		svc = func(namespace string, name string, clusterIPs []string) *corev1.Service {
+			return &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "kube-system",
+					Name:      "kube-dns",
+				},
+				Spec: corev1.ServiceSpec{
+					ClusterIPs: clusterIPs,
+				},
+			}
+		}
+		parseCIDR = func(cidr string) *net.IPNet {
+			_, parsedCIDR, err := net.ParseCIDR(cidr)
+			Expect(err).ToNot(HaveOccurred())
+			return parsedCIDR
+		}
+	)
+	DescribeTable("composing dhcp options", func(t dhcpTest) {
+		svcs := []corev1.Service{}
+		if t.dns != nil {
+			svcs = append(svcs, *t.dns)
+		}
+		fakeClient := fake.NewSimpleClientset(&corev1.ServiceList{
+			Items: svcs,
+		})
+		cidrs := []*net.IPNet{}
+		for _, cidr := range t.cidrs {
+			cidrs = append(cidrs, parseCIDR(cidr))
+		}
+		ipv4Options, ipv6Options, err := ComposeDHCPOptionsPair(fakeClient, t.hostname, cidrs)
+		if t.hasError {
+			Expect(err).To(HaveOccurred())
+		} else {
+			Expect(err).ToNot(HaveOccurred())
+		}
+		Expect(ipv4Options).To(Equal(t.expectedIPv4Options))
+		Expect(ipv6Options).To(Equal(t.expectedIPv6Options))
+	},
+		Entry("IPv4 Single stack and k8s dns", dhcpTest{
+			cidrs:               []string{"192.168.25.0/24"},
+			hostname:            "foo1",
+			dns:                 svc("kube-system", "core-dns", []string{"192.167.23.44"}),
+			expectedIPv4Options: ComposeDHCPv4Options("192.168.25.0/24", ARPProxyIPv4, "192.167.23.44", "foo1"),
+		}),
+		Entry("IPv6 Single stack and k8s dns", dhcpTest{
```

Comment:
This is before we took that decission, but still is not bad that we keep partial support at some parts, so when we implement it in the future this will be less stuff to do.